fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -152,32 +149,30 @@ public FDBDirectory(@Nonnull Subspace subspace, @Nonnull FDBRecordContext contex
      * @return current increment value
      */
     public synchronized long getIncrement() {
-        increment(LuceneEvents .Counts.LUCENE_GET_INCREMENT_CALLS);
-        return context.ensureActive().get(sequenceSubspaceKey).thenApply(
-            (value) -> {
-                if (value == null) {
-                    context.ensureActive().set(sequenceSubspaceKey, Tuple.from(1L).pack());
-                    return 1L;
-                } else {
-                    long sequence = Tuple.fromBytes(value).getLong(0) + 1;
-                    context.ensureActive().set(sequenceSubspaceKey, Tuple.from(sequence).pack());
-                    return sequence;
-                }
-            }).join();
+        increment(LuceneEvents.Counts.LUCENE_GET_INCREMENT_CALLS);
+        CompletableFuture<Long> future = context.ensureActive().get(sequenceSubspaceKey).thenApply(value -> {
+            if (value == null) {
+                context.ensureActive().set(sequenceSubspaceKey, Tuple.from(1L).pack());
+                return 1L;
+            } else {
+                long sequence = Tuple.fromBytes(value).getLong(0) + 1;
+                context.ensureActive().set(sequenceSubspaceKey, Tuple.from(sequence).pack());
+                return sequence;
+            }
+        });
+        return Objects.requireNonNull(context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_GET_INCREMENT, future));
     }
 
     private void increment(StoreTimer.Count counter) {
-        final FDBStoreTimer timer = context.getTimer();
-        if (timer != null) {
-            timer.increment(counter);
-        }
+        context.increment(counter);
```

Comment:
As a note here: the `FDBRecordContext::increment` method that this is replacing already checks to see if the timer is null (and updates the counter if the timer is there), so this change is to remove the duplication:

https://github.com/FoundationDB/fdb-record-layer/blob/21baa55b2f33f98fb1dfafaf5794f0bf2a583733/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBTransactionContext.java#L137-L141

Same for the other `increment` and `record` methods that are updated here.